src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ClusterApplierServiceStatsCollector.java

```diff
@@ -0,0 +1,164 @@
+                        "Skipping ClusterApplierServiceStatsCollector");
+                return;
+            }
+
+            ClusterApplierServiceMetrics clusterApplierServiceMetrics = new ClusterApplierServiceMetrics(
+                    computeLatency(clusterApplierServiceStats), computeFailure(clusterApplierServiceStats));
+
+            value.setLength(0);
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                    .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            value.append(clusterApplierServiceMetrics.serialize());
+            saveMetricValues(value.toString(), startTime);
+            this.tracker = new MetricsTracker(clusterApplierServiceStats.timeTakenInMillis,
+                    clusterApplierServiceStats.failedCount, clusterApplierServiceStats.totalCount);
+            PerformanceAnalyzerApp.WRITER_METRICS_AGGREGATOR.updateStat(
+                    WriterMetrics.CLUSTER_APPLIER_SERVICE_STATS_COLLECTOR_EXECUTION_TIME, "",
+                    System.currentTimeMillis() - mCurrT);
+        } catch (Exception ex) {
+            PerformanceAnalyzerApp.ERRORS_AND_EXCEPTIONS_AGGREGATOR.updateStat(
+                    ExceptionsAndErrors.CLUSTER_APPLIER_SERVICE_STATS_COLLECTOR_ERROR, "", 1);
+            LOG.debug("Exception in Collecting Cluster Applier Service Metrics: {} for startTime {}",
+                    () -> ex.toString(), () -> startTime);
+        }
+    }
+
+    private Object getClusterApplierServiceStats() throws InvocationTargetException, IllegalAccessException,
+            NoSuchMethodException {
+        Method method = ClusterApplierService.class.getMethod(GET_CLUSTER_APPLIER_SERVICE_STATS_METHOD_NAME);
+        return method.invoke(ESResources.INSTANCE.getClusterService().getClusterApplierService());
+    }
+
+    private long computeLatency(final ClusterApplierServiceStats currentMetrics) {
+        final long rate = computeRate(tracker.getPrevTotalCount());
+        if(rate == 0) {
+            return 0L;
+        }
+        return (currentMetrics.timeTakenInMillis - tracker.getPrevTimeTakenInMillis()) / rate;
+    }
+
+    private long computeRate(final long currentTotalCount) {
+        return currentTotalCount - tracker.getPrevTotalCount();
+    }
+
+    private long computeFailure(final ClusterApplierServiceStats currentMetrics) {
+        return currentMetrics.failedCount - tracker.getPrevFailedCount();
+    }
+
+    @Override
+    public String getMetricsPath(long startTime, String... keysPath) {
+        if (keysPath.length != KEYS_PATH_LENGTH) {
+            throw new RuntimeException("keys length should be " + KEYS_PATH_LENGTH);
+        }
+        return PerformanceAnalyzerMetrics.generatePath(startTime, PerformanceAnalyzerMetrics.sClusterApplierService);
+    }
+
+    public static class ClusterApplierServiceStats  {
+        private long totalCount;
+        private long timeTakenInMillis;
```

Comment:
The ClusterApplier thread could finish in sub milisecond, do you want to keep the data type as double instead of long?

Reply:
Yes. I already changed that. Will publish it in the next reveision.